File: 백준/Gold/2447. 별 찍기 － 10/별 찍기 － 10.py
import sys
import math

def print_starmap(k):

    if k == 3:
        return [
            ['*', '*', '*'],
            ['*', ' ', '*'],
            ['*', '*', '*']
        ]

    itemStarMap = print_starmap(k // 3)
    newStarMap = []

    idxCnt = k // 3
    for i in range(k):
        idxArr = itemStarMap[i%idxCnt][:]
        if i // idxCnt != 1:
            inputArr = idxArr[:] * 3
            newStarMap.append(inputArr)
        else:
            inputArr = idxArr[:]
            inputArr += [' ']*idxCnt
            inputArr += idxArr[:]
            newStarMap.append(inputArr)

    return newStarMap


n = int(sys.stdin.readline())

starmap = print_starmap(n)

for arr in starmap:
    print(''.join(map(str, arr)))

File: 백준/Gold/2447. 별 찍기 － 10/별 찍기 － 10.js
const fs = require("fs");
const inputs = fs.readFileSync("/dev/stdin").toString().trim()

function printStarmap(k) {
    if(k === 3) {
        return [
            ['*', '*', '*'],
            ['*', ' ', '*'],
            ['*', '*', '*']
        ]
    }
    
    const itemStarMap = printStarmap(Math.floor(k / 3))
    let newStarMap = []
    
    const idxCnt = Math.floor(k / 3)
    for(let i = 0; i < k; i++) {
        let idxArr = itemStarMap[i % idxCnt]
        
        if(Math.floor(i / idxCnt) !== 1) {
            newStarMap.push([...idxArr, ...idxArr, ...idxArr])
        }
        else {
            let inputArr = [
                ...idxArr,
                ...Array.from({length: idxCnt}, () => ' '),
                ...idxArr
            ]
            newStarMap.push(inputArr)
        }
    }
    
    return newStarMap
}

const n = parseInt(inputs)

const starMap = printStarmap(n)

for(const arr of starMap) {
    console.log(arr.join(''))
}
